mlflow/server/js/src/experiment-tracking/components/CompareRunView.js

```diff
@@ -37,6 +37,40 @@ export class CompareRunView extends Component {
     intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }).isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      tableContainerWidth: null,
+      collapseParamBlock: false,
+      collapseMetricBlock: false,
+      onlyShowParamDiff: false,
+      onlyShowMetricDiff: false,
+    };
+    // onResizeHandler need access `this.state`, so bind `this`.
+    this.onResizeHandler = this.onResizeHandler.bind(this);
+    this.onTableBlockScrollHandler = this.onTableBlockScrollHandler.bind(this);
+
+    this.tableContainerRef = React.createRef();
+  }
+
+  onResizeHandler(e) {
+    const container = this.tableContainerRef.current;
+    if (container !== null) {
+      const containerWidth = container.clientWidth;
+      this.setState({ tableContainerWidth: containerWidth });
+    }
+  }
+
+  onTableBlockScrollHandler(e) {
+    const blocks = this.tableContainerRef.current.querySelectorAll('.compare-table .table-block');
```

Comment:
This is the only place I preserve the "querySelectorAll".
Because:
* We cannot set scrollbar offset via CSS. So we have to put it in the onScroll handler.
* We can use multiple react Refs to holds these table-blocks, but it make code more complex, using the selector might be the easiest way.
* I limit the `querySelectorAll` to select inside the `this.tableContainerRef` element so it will be safe.